mesheryctl/internal/cli/root/system/start.go

```diff
@@ -194,11 +194,9 @@ func start() error {
 	switch currPlatform {
 	case "docker":
 
-		if _, err := os.Stat(utils.MesheryFolder); os.IsNotExist(err) {
-			// download the docker-compose.yaml file corresponding to the current version
-			if _, err := utils.DownloadDockerComposeFile(currCtx, true); err != nil {
-				return errors.Wrapf(err, utils.SystemError(fmt.Sprintf("failed to download %s file", utils.DockerComposeFile)))
-			}
+		// download the docker-compose.yaml file corresponding to the current version
+		if _, err := utils.DownloadDockerComposeFile(currCtx, true); err != nil {
+			return errors.Wrapf(err, utils.SystemError(fmt.Sprintf("failed to download %s file", utils.DockerComposeFile)))
```

Comment:
This update was done to ensure the correct version of the docker-compose file is downloaded based on context.
For example:
suppose I already have a file `meshery.yaml` that was downloaded when my context used `channel: edge`,
now I change my context such that it uses `channel: stable` . 
There is no indicator in `meshery.yaml`  that can identify the version of the file.

If this is acceptable, then we can also remove `ValidateComposeFileForRecreation` function as it becomes redundant.

@anirudhjain75 @navendu-pottekkat

Reply:
So the meshery.yaml is to be fetched only when there is a change in version. The reasonable way would be to compare the latest releasr version and see it there is a change... The compose is to be built using context, a function can be created so as to update the meshery.yaml based off on context

Reply:
Is this an optimization that can be addressed later?